### PR TITLE
finish run to last pos + get involved modal + no-navigate

### PR DESCRIPTION
**Pull request by trilopin**
#### Summary
Finish run has been moved to the end of sections

Get involved when the channel is public changes:
- now also triggers a confirmation modal 
- do not navigate to the channel

If this pr is merged after https://github.com/mattermost/mattermost-plugin-playbooks/pull/1309, some extra effort will be needed.
 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45695
https://mattermost.atlassian.net/browse/MM-45696

#### Checklist
- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

**Comment by trilopin** [comment]
@justinegeffen here you have the last case I mentioned to you: public channel and get-involved also trigger a modal. 

I kept the same title with different modal content and toast. If you also want a different title please let me know.

**Review by crspeller — APPROVED** [review]

**Review by agarciamontoro — APPROVED** [review]
Looks great, thank you!

File: webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx [hunk 1]
```diff
@@ -48,33 +47,28 @@ export const RunHeader = ({playbookRun, playbookRunMetadata, role, onViewInfo, o
...
-            addToast(formatMessage({defaultMessage: 'Request has been sent to the run channel.'}), ToastType.Success);
-        }
-    };
-
-    const navigateToChannel = () => {
-        if (!playbookRunMetadata) {
-            return;
-        }
-        navigateToUrl(`/${playbookRunMetadata.team_name}/channels/${playbookRunMetadata.channel_name}`);
+        addToast(formatMessage({defaultMessage: 'You just joined this run.'}), ToastType.Success);
```
Comment (justinegeffen) [inline webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx:71]:
```suggestion
        addToast(formatMessage({defaultMessage: 'You've joined this run.'}), ToastType.Success);
```

File: webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx [hunk 2]
```diff
@@ -127,7 +121,8 @@ export const RunHeader = ({playbookRun, playbookRunMetadata, role, onViewInfo, o
             <ConfirmModal
                 show={showGetInvolvedConfirm}
                 title={formatMessage({defaultMessage: 'Confirm get involved'})}
-                message={formatMessage({defaultMessage: 'A message will be sent to the run channel, requesting them to add you as a participant.'})}
+                message={channel === null ? formatMessage({defaultMessage: 'A message will be sent to the run channel, requesting them to add you as a participant.'}) : formatMessage({defaultMessage: 'You are about to join this run.'})
```
Comment (justinegeffen) [inline webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx:124]:
```suggestion
                message={channel === null ? formatMessage({defaultMessage: 'Your participation request will be sent to the run channel.'}) : formatMessage({defaultMessage: 'You're about to join this run.'})
```

**Review by justinegeffen — COMMENTED** [review]
Thanks, @trilopin! I've added some suggestions and also a question around the error message.

File: webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx [hunk 3]
```diff
@@ -48,33 +47,28 @@ export const RunHeader = ({playbookRun, playbookRunMetadata, role, onViewInfo, o
...
+        }
 
         // Channel null value comes from error response (and we assume that is mostly 403)
         // If we don't have access to channel we'll send a request to be added,
         // otherwise we directly join it
         if (channel === null) {
-            setShowGetInvolvedConfirm(true);
+            const response = await requestGetInvolved(playbookRun.id);
+            if (response?.error) {
+                addToast(formatMessage({defaultMessage: 'It was not possible to request to get involved'}), ToastType.Failure);
```
[the remark this answers is not in the document]
Comment (trilopin) [reply webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx:64]: This message is shown when something unexpected happens, like a user manipulating requests or development mistakes. I think there's nothing we can recommend to the user.

**Pull request merged**